autoPyTorch/pipeline/components/setup/network_backbone/utils.py

```diff
@@ -92,15 +92,31 @@ def backward(ctx: typing.Any,
 shake_drop = ShakeDropFunction.apply
 
 
-def shake_get_alpha_beta(is_training: bool, is_cuda: bool
-                         ) -> typing.Tuple[torch.tensor, torch.tensor]:
+def shake_get_alpha_beta(
+        is_training: bool,
+        is_cuda: bool,
+        method: str
+) -> typing.Tuple[torch.tensor, torch.tensor]:
+
     if not is_training:
         result = (torch.FloatTensor([0.5]), torch.FloatTensor([0.5]))
         return result if not is_cuda else (result[0].cuda(), result[1].cuda())
 
     # TODO implement other update methods
-    alpha = torch.rand(1)
-    beta = torch.rand(1)
+    if method != 'even-even':
+        alpha = torch.rand(1)
+    else:
+        alpha = torch.FloatTensor([0.5])
+    if method in ['shake-shake', 'shake-drop']:
```

Comment:
space between `if` statements
```suggestion
    else:
        alpha = torch.FloatTensor([0.5])

    if method in ['shake-shake', 'shake-drop']:
```

Reply:
on that matter, `shake-drop` will only have random alpha and beta, if we really want to have the search space of the paper, we should pass even, even also for shake-drop.

Reply:
okay, I can make shake_shake_methods to be dependent on both shake-shake and shake-drop. and add even-even, shake-shake, shake-even, m3 as the methods for alpha and beta. Does that make sense?

Reply:
Yes it does for the moment.

Reply:
yeah I added them